File: 682. Baseball Game/682. Baseball Game.java
import java.util.Stack;

class Solution {
    public int calPoints(String[] operations) {
        Stack<Integer> stack = new Stack<>();

        for (String op : operations) {
            if (op.equals("+")) {
                stack.push(stack.peek() + stack.get(stack.size() - 2));
            } else if (op.equals("D")) {
                stack.push(stack.peek() * 2);
            } else if (op.equals("C")) {
                stack.pop();
            } else {
                stack.push(Integer.parseInt(op));
            }
        }

        int summ = 0;
        while (!stack.isEmpty()) {
            summ += stack.pop();
        }
        return summ;
    }
}

File: 682. Baseball Game/682. Baseball Game.js
/**
 * @param {string[]} operations
 * @return {number}
 */
var calPoints = function (operations) {
  const stack = [];

  for (const op of operations) {
    if (op === "+") {
      stack.push(stack.at(-1) + stack.at(-2));
    } else if (op === "D") {
      stack.push(stack.at(-1) * 2);
    } else if (op === "C") {
      stack.pop();
    } else {
      stack.push(Number(op));
    }
  }

  return stack.reduce((a, c) => a + c, 0);
};
